DSpace/DSpace
# Remove LCNameDataProvider and config from DSpace 7 REST API

## References
* Fixes https://github.com/DSpace/DSpace/issues/3019

## Description
This PR removes the LCNameDataProvider and it's configuration from DSpace 7 as the service has been discontinued.

## Instructions for Reviewers

List of changes in this PR:
* Removed the LCNameDataProvider from DSpace 7
* Removed the config for it from DSpace 7

How to test this PR:
* Build the code and deploy this branch
* Go to the `server/api/integration/externalsources` endpoint and verify that the LCNameData external sources endpoint is not available

## Checklist
